### DISCUSSION TITLE
Checksum

### DISCUSSION THREAD

**Discussion by yusufKilicarslan** [ANLAB-KAIST/KENSv3#77] [Project Q&A 2. Transport Layer (TCP)]
Could you tell me what is wrong with this code:
Calculated checksum and sent checksum are never equal (never calls exit(1)).

<img width="457" alt="image" src="https://user-images.githubusercontent.com/21026320/162798790-b560ad2e-694e-4d33-9c13-61287ce013f5.png">

**Comment by Pusnow** [comment]
You should compare it with the original checksum value (before `tcp[16] = tcp[17] = 0;`).
Also, you should apply network ordering and inverse (`htons(~NetworkUtil::tcp_sum(src,dest,tcp, 20))`). 

Or, remove `tcp[16] = tcp[17] = 0;` and compare the return value of `tcp_sum` with `0xFF`.
